pygmt/src/meca.py

```diff
@@ -0,0 +1,413 @@
+          n_exponent, n_azimuth, n_plunge, p_exponent, p_azimuth, p_plunge,
+          exponent*
+
+    scale: str
+        Adjusts the scaling of the radius of the beachball, which is
+        proportional to the magnitude. Scale defines the size for
+        magnitude = 5 (i.e. scalar seismic moment M0 = 4.0E23 dynes-cm)
+    longitude: int, float, list, or 1d numpy array
+        Longitude(s) of event location. Ignored if `spec` is not a
+        dictionary. List must be the length of the number of events.
+        Ignored if `spec` is a DataFrame and contains a 'longitude' column.
+    latitude: int, float, list, or 1d numpy array
+        Latitude(s) of event location. Ignored if `spec` is not a
+        dictionary. List must be the length of the number of events.
+        Ignored if `spec` is a DataFrame and contains a 'latitude' column.
+    depth: int, float, list, or 1d numpy array
+        Depth(s) of event location in kilometers. Ignored if `spec` is
+        not a dictionary. List must be the length of the number of events.
+        Ignored if `spec` is a DataFrame and contains a 'depth' column.
+    convention: str
+        ``"aki"`` (Aki & Richards), ``"gcmt"`` (global CMT), ``"mt"``
+        (seismic moment tensor), ``"partial"`` (partial focal mechanism),
+        or ``"principal_axis"`` (principal axis). Ignored if `spec` is a
+        dictionary or dataframe.
+    component: str
+        The component of the seismic moment tensor to plot. ``"full"`` (the
+        full seismic moment tensor), ``"dc"`` (the closest double couple
+        with zero trace and zero determinant), ``"deviatoric"`` (zero
+        trace)
+    plot_longitude: int, float, list, or 1d numpy array
+        Longitude(s) at which to place beachball, only used if `spec` is a
+        dictionary. List must be the length of the number of events.
+        Ignored if `spec` is a DataFrame and contains a 'plot_longitude'
+        column.
+    plot_latitude: int, float, list, or 1d numpy array
+        Latitude(s) at which to place beachball, only used if `spec` is a
+        dictionary. List must be the length of the number of events.
+        Ignored if `spec` is a DataFrame and contains a 'plot_latitude'
+        column.
+    offset: bool or str
+        Offsets beachballs to the longitude, latitude specified in
+        the last two columns of the input file or array,
+        or by `plot_longitude` and `plot_latitude` if provided. A small
+        circle is plotted at the initial location and a line connects
+        the beachball to the circle. Specify pen and optionally append
+        ``+ssize`` to change the line style and/or size of the circle.
+    no_clip : bool
+        Does NOT skip symbols that fall outside frame boundary specified by
+        *region* [Default is False, i.e. plot symbols inside map frame
+        only].
+    {J}
+    {R}
+    {B}
+    {V}
+    {XY}
+    {p}
+    {t}
+    """
+
```

Comment:
Note that `fig.meca` doesn't appear to use `self._preprocess` like the other `fig` modules, but probably doesn't matter according to https://github.com/GenericMappingTools/pygmt/issues/685#issuecomment-723533013?

Reply:
Seems OK to me.